### PR TITLE
[Doc] Fix Pulsar c++ client windows build docs

### PR DESCRIPTION
**Pull request by gaoran10**
### Motivation

Currently, docs that the Pulsar c++ client windows building-related docs are not complete.

### Modifications

Add Pulsar c++ client windows building steps in Apache Pulsar website.
Add Pulsar c++ client windows building result check section.

### CODE REVIEW

File: pulsar-client-cpp/README.md [hunk 1]
```diff
@@ -235,6 +235,26 @@ ${PULSAR_PATH}/pulsar-client-cpp/cmake -DPROTOC_PATH=C:/protobuf/bin/protoc -DCM
...
+```
+${PULSAR_PATH}/pulsar-client-cpp/build/lib/Release/pulsar.lib
+${PULSAR_PATH}/pulsar-client-cpp/build/lib/Release/pulsar.dll
+```
+
+#### Examples
+
+##### Add windows environment paths
+```
+${PULSAR_PAHT}/pulsar-client-cpp/build/lib/Release
```
Comment (aahmed-se) [inline pulsar-client-cpp/README.md:250]: Fix path spelling

Comment (gaoran10) [reply pulsar-client-cpp/README.md]: I'll fix it, thanks.

File: pulsar-client-cpp/README.md [hunk 2]
```diff
@@ -235,6 +235,26 @@ ${PULSAR_PATH}/pulsar-client-cpp/cmake -DPROTOC_PATH=C:/protobuf/bin/protoc -DCM
 #This will generate pulsar-cpp.sln. Open this in Visual Studio and build the desired configurations.
 ```
 
+#### Checks
+
+##### Client library will be placed in
```
Comment (Anonymitaet) [inline pulsar-client-cpp/README.md:240]:
```suggestion
##### Client libraries are available in the following places.
```

File: pulsar-client-cpp/README.md [hunk 3]
```diff
@@ -235,6 +235,26 @@ ${PULSAR_PATH}/pulsar-client-cpp/cmake -DPROTOC_PATH=C:/protobuf/bin/protoc -DCM
...
+
+#### Examples
+
+##### Add windows environment paths
+```
+${PULSAR_PATH}/pulsar-client-cpp/build/lib/Release
+${PULSAR_PATH}/pulsar-client-cpp/vcpkg_installed
+```
+
+##### Examples will be placed in
```
Comment (Anonymitaet) [inline pulsar-client-cpp/README.md:254]:
```suggestion
##### Examples are available in
```

File: site2/docs/client-libraries-cpp.md [hunk 4]
```diff
@@ -234,6 +234,37 @@ Pulsar releases are available in the [Homebrew](https://brew.sh/) core repositor
...
+```
+
+2. Install all necessary dependencies.
+
+```shell
+cd ${PULSAR_HOME}/pulsar-client-cpp
+vcpkg install --feature-flags=manifests --triplet x64-windows
+```
+
+3. Build c++ libraries.
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-cpp.md:254]:
```suggestion
3. Build C++ libraries.
```

File: site2/docs/client-libraries-cpp.md [hunk 5]
```diff
@@ -234,6 +234,37 @@ Pulsar releases are available in the [Homebrew](https://brew.sh/) core repositor
...
+
+```shell
+cmake -B ./build -A x64 -DBUILD_PYTHON_WRAPPER=OFF -DBUILD_TESTS=OFF -DVCPKG_TRIPLET=x64-windows -DCMAKE_BUILD_TYPE=Release -S .
+cmake --build ./build --config Release
+```
+
+4. Client library will be placed in.
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-cpp.md:261]:
```suggestion
4. Client libraries are available in the following places.
```

**Review by BewareMyPower — APPROVED** [review]

**Review by Anonymitaet — APPROVED** [review]

**Comment by Anonymitaet** [comment]
Thanks for your contribution. Does this affect only master or other versioned docs? 
If latter, could you please help update all affected versions? Thanks

**Review by sijie — APPROVED** [review]

**Pull request merged**